### ISSUE TITLE
SC4S not starting if SC4S_LISTEN_LEEF_TCP_PORT is set

### ISSUE THREAD

**Issue by davidandreatta** [splunk/splunk-connect-for-syslog#1936]
Hello, 

SC4S version v2.42.0 won't start when setting SC4S_LISTEN_LEEF_TCP_PORT.

```
sc4s version=2.42.0
starting goss
Traceback (most recent call last):
  File "/etc/syslog-ng/conf.d/sources/source_syslog/plugin.py", line 106, in <module>
    product = port_parts[1].lower()
IndexError: list index out of range
[2022-11-23T11:08:10.302197] confgen: Generator program returned with non-zero exit code; reference='/etc/syslog-ng/conf.d/sources/source_syslog/plugin.conf:4:1', context='root', block='source_syslog', exec='/etc/syslog-ng/conf.d/sources/source_syslog/plugin.py', rc='256'
Error parsing config, syntax error, unexpected LL_ERROR, expecting end of file in /etc/syslog-ng/conf.d/sources/source_syslog/plugin.conf:4:1-4:14:
Included from /etc/syslog-ng/syslog-ng.conf:44:1-44:1:
39
40      @include "conf.d/plugin/*.conf"
41
42      @include "conf.d/sources/*.conf"
43      @include "conf.d/sources/*/*.conf"
44----> @include "conf.d/local/config/sources/*.conf"
44----> ^
45      @include "conf.d/local/config/sources/*/*.conf"
46
47      @include "conf.d/destinations/*.conf"
48      @include "conf.d/destinations/*/*.conf"
49
```

In plugin.py the problem is at line 103, if port_parts has lenght of 1 (which is the case of SC4S_LISTEN_LEEF_TCP_PORT) it produces an index out of range:
```
     port_parts = port_id.split('_',maxsplit=3)
        if len(port_parts)<4:
            vendor = port_parts[0].lower()
            product = port_parts[1].lower()
```

Thanks

**Comment by rjha-splunk** [comment]
Thanks for reporting it , the release will be available shortly.version 2.42.1

**Comment by rjha-splunk** [comment]
The fix is released in 2.42.1.
